Sources/SwifterSwift/SwiftStdlib/ArrayExtensions.swift

```diff
@@ -33,6 +33,24 @@ public extension Array {
         guard startIndex..<endIndex ~= otherIndex else { return }
         swapAt(index, otherIndex)
     }
+
+    /// SwifterSwift: Sort an array like another array based on a keyPath. If the other array doesn't contain a certain value, it will be sorted last.
+    ///
+    ///        [MyStruct(x: 3), MyStruct(x: 1), MyStruct(x: 2)].sorted(like: [1, 2, 3], keyPath: \.x)
+    ///            -> [MyStruct(x: 1), MyStruct(x: 2), MyStruct(x: 3)]
+    ///
+    /// - Parameters:
+    ///   - otherArray: array containing elements in the desired order.
+    ///   - keyPath: keyPath indiciating the property that the array should be sorted by
+    /// - Returns: sorted array.
+    func sorted<T: Hashable>(like otherArray: [T], keyPath: KeyPath<Element, T>) -> [Element] {
```

Comment:
This is an interesting method, seems very specific tho... could you provide an practical use case for this?

Reply:
I have a list of CloudKit record names (Strings). I query CloudKit for the records, but they are returned in random order. This method can be used to sort the retrieved records based on the order of my record names. This principle can also be applied to database queries.

Reply:
Hummm... thanks for the response, but why just sort by name would not be sufficient?

Reply:
Because I want to achieve an order like in the other array that can't be expressed by a sort descriptor. E.g the order [2, 3, 1] can't be expressed using sort descriptors.